labs/a8.md

```diff
@@ -0,0 +1,736 @@
+can be reasonably sure that this file, and only this file, will always hash to 
+that particular value. This means, if you want to verify that the file you 
+downloaded hasn't been corrupted, you can simply compare the hashes (or more 
+specifically, a signature over the hash) to ensure that the file you've 
+downloaded hasn't been tampered with.
+
+### Encryption Lab Activity
+
+In this lab, we will be exploring the topics covered so far: symmetric 
+encryption and public key cryptography, hashing, signatures, and certificates. 
+Please make sure you are logged into the appropriate server over ssh prior to 
+completing any of the following tasks.
+
+##### Introduction: Making SSH keys
+
+We'll begin our exploration of security building blocks by creating SSH 
+keypairs for ourselves, in case you haven't already. If you already have an SSH 
+key on your personal computer that you would like to continue to use, or one on 
+`tsunami`, feel free to skip these steps and proceed to task 1, but please make 
+sure that you can use that key to log into your student VM.
+
+As mentioned earlier in this document, and in [lab 4](labs/a4), here is how to
+create a 4096-bit RSA SSH keypair.
+
+1. Log into `tsunami`. It is important you do these steps on `tsunami` and not 
+your laptop, an OCF desktop, or your student VM.
+2. Check to make sure you do not have an existing SSH key:
+    a. Enter `ls -l ~/.ssh`
+    b. If you see files like `id_rsa` or `id_rsa.pub` then you already have an 
+SSH key.
+3. If no key is present, create a new key now by typing `ssh-keygen -t rsa -b 
+4096`, and hit enter when prompted.
+4. If you are asked to type a password to protect the key, you may choose to 
+enter one, but for the purpose of this lab, it is advisable to skip doing so.
+
+You should now see two new files in the `~/.ssh/` directory. `id_rsa` is your 
+SSH private key, and it should like the following:
+
+    $ cat ~/.ssh/id_rsa 
+    -----BEGIN RSA PRIVATE KEY-----
+    <random characters>
+    -----END RSA PRIVATE KEY------
+    
+Please try to not lose or leak this private key. The public key, `id_rsa.pub`, 
+should be much shorter and look like the following:
+
+    $ cat ~/.ssh/id_rsa.pub
+    ssh-rsa <characters> <username>@<host>
+    
+It is safe to transfer your RSA public key out in the open, whereas you should 
+be extremely careful when doing the same with your private key.
+
+You've just created for yourself an RSA keypair suitable for SSH authentication.
+
+For the remaining tasks in this lab, you will be tasked with figuring out the 
+appropriate commands yourself, by Googling, reading man pages, etc. Hints will 
+be given as footnotes as necessary. Please document the commands you use and be 
+prepared to provide a rationale for why you believe those commands are correct.
+
+###### Task 1
+
+Figure out how to use the SSH key you've created to log into your student 
+VM.[^ssh-copy-id]
```

Comment:
I think the formatting got messed up here.

Reply:
the line was broken at VM to avoid the footnote from being mispositioned. it's kinda ugly though, yeah